### DISCUSSION TITLE
Enriching records in custom JS server-side code

### DISCUSSION THREAD

**Discussion by ParisZX** [pocketbase/pocketbase#3846] [Q&A] [answered]
Hi, I was wondering if anyone tried to implement any scenario similar to this and run into any issues:

I have a `users` collection and a `submissions` collection that has a users relation field called `user`. The `submissions` collection also has a relation `form` (single), that points to the `forms` collection and a relation `values` (multiple) that points to the `values` collection.

I was trying to create a custom endpoint using pb_hooks and the embedded js engine, and ran into the following issue: 

- When I try to use enrichRecords with `submissions(user)` as the expand string, it works just fine
- When I try `submissions(user).form`, it also works as expected
- When I try `submissions(user).values`, it returns an error `GoError: Failed to expand: map[submissions(user).values:SQL logic error: too many SQL variables (1)]`, which I guess has to do with the fact that this is a `multiple` relation, but the weird thing is that when I use the same expand string through the client JS library, it works just fine. Is there any workaround to that that doesn't include iterating through the `values` ids and getting them one by one?

**Comment by ganigeorgiev** [comment, marked as answer]
This looks more like a bug but I'm not able to reproduce it with my test database.

Could you provide a code sample with your custom JS action and attach your collections configuration as json (_Admin UI > Settings > Export collections_)?

**Reply by ParisZX** [reply]
Thanks for the response, I've been trying since yesterday to recreate the issue in a local test instance so that I can send you more details, but apparently I can't recreate it either, as of yet. So, it's probably something more complex than that, that creates the issue. Let me get back to you as soon as I have more details and an example export.

**Reply by ParisZX** [reply]
As much as I tried I couldn't reproduce it locally and, as part of some tests, I have accidentally deleted the instance where this originally happened. The more recent versions of the database have the same collections and relations I mentioned before, with only a few other differences (rules, extra fields etc.), but the error never re-appeared. I guess I originally either somehow messed up the code (although the exact same snippet was running fine if I removed the `.values` part) or there was some other kind of issue... Is there any chance it had to do with rules over the collections?

In any case, I ended up using the `expandRecords` method in the final code and it works just fine, but `enrichRecords` is also working as expected. I'll close this discussion as it's probably a false alarm! :)

**Reply by ganigeorgiev** [reply]
> I ended up using the `expandRecords` method in the final code

That's OK but note that [`expandRecords`](https://pocketbase.io/jsvm/classes/Dao.html#expandRecords) if non-explicit "rel fetch" function is specified it will fetch all expand relation records without checking the API rules (_for checking API rules you'll need a request info context, which `enrichRecords` extracts from its first argument_).

I'll investigate it in more details this weekend and will try to reproduce the reported error.
